### ISSUE TITLE
Implement Column Click Sorting

### ISSUE THREAD

**Issue by djjd47130** [djjd47130/GitHubBackup#26]
If user clicks on a column header, sort by that column. If user clicks same column header again, toggle between ascending / descending.

**Comment by djjd47130** [comment]
This needs a bit more attention, specifically a more robust column manager. I'm afraid to roll out such a thing for a few reasons, including the fact that I plan to migrate to a Virtual TreeView, which works completely different. I was about to use the existing constants for the column indices, but then realized those constants don't correspond with the actual columns in the list. So I need the facility to detect the corresponding field per column. Those constants aren't for the column index - it's for the actual fields for sorting - some of which aren't even visible in the actual list. 

Long story short, before attacking this, I'd like to first implement user-customizable column settings, allowing users to choose which columns to show, their widths, and their position.
